### PR TITLE
Adapted README file, initialize api documentation

### PR DESCRIPTION
**Pull request by jlaehne**
### Description of the change
Adapted the `README` file in the main directory to the new structure of plugins -> rename to `CONTRIBUTING.rst` and link from user guide.

Initialize a page in the documentation for the API.

### Progress of the PR
- [X] Change implemented (can be split into several points),
- [x] add API to documentation,
- [x] create separate `CONTRIBUTING.rst` file,
- [x] check formatting,
- [X] ready for review.

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+The **dictionary** contains the following fields:
+* `'data'` -- multidimensional numpy array
+* `'axes'` -- list of dictionaries describing the axes containing the fields
+  `'name'`, `'units'`, `'index_in_array'`, and
+  - either `'size'`, `'offset'`, and `'scale'`
+  - or a numpy array `'axis'` containing the full axes vector
+* `'metadata'` -- dictionary containing the parsed metadata
+* `'original_metadata'` -- dictionary containing the full metadata tree from the
+  input file
```
Comment (ericpre) [inline docs/api.rst:28]:
```suggestion
The **dictionary** contains the following fields:

* ``'data'`` -- multidimensional numpy array
* ``'axes'`` -- list of dictionaries describing the axes containing the fields
  ``'name'``, ``'units'``, ``'index_in_array'``, and
  
  - either ``'size'``, ``'offset'``, and ``'scale'``
  - or a numpy array ``'axis'`` containing the full axes vector

* ``'metadata'`` -- dictionary containing the parsed metadata
* ``'original_metadata'`` -- dictionary containing the full metadata tree from the
  input file
```

File: docs/api.rst [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    format_name: <String>
+    description: <String>
+    full_support: <Bool>	# Whether all the Hyperspy features are supported
+    # Recognised file extension
+    file_extensions: <Tuple of string>
+    default_extension: <Int>	# Index of the extension that will be used by default
+    # Writing capabilities
+    writes: <Bool>
+    # Support for non-uniform axis
+    non_uniform_axis = <Bool>
```
Comment (ericpre) [inline docs/api.rst:90]:
```suggestion
  .. code-block:: yaml

      format_name: <String>
      description: <String>
      full_support: <Bool>	# Whether all the Hyperspy features are supported
      # Recognised file extension
      file_extensions: <Tuple of string>
      default_extension: <Int>	# Index of the extension that will be used by default
      # Writing capabilities
      writes: <Bool>
      # Support for non-uniform axis
      non_uniform_axis = <Bool>
```

File: docs/api.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+* ``api.py`` -- Python file that implements the actual reader. The IO functionality
+  should be interfaced with the following functions:
+
+      * A function called ``file_reader`` with at least one attribute: ``filename``
+
+      * (optional) A function called ``file_writer`` with at least two attributes: 
+        ``filename`` and ``object2save`` (a python dictionary) in that order.
```
Comment (ericpre) [inline docs/api.rst:98]:
```suggestion
  * A function called ``file_reader`` with at least one attribute: ``filename``
  * (optional) A function called ``file_writer`` with at least two attributes: 
    ``filename`` and ``object2save`` (a python dictionary) in that order.
```

**Comment by jlaehne** [comment]
Do we still need the [rsciio/README](https://github.com/hyperspy/rosettasciio/pull/31/files#diff-85e6718f5a3e1ff19e9a45f0d6d7752d0c3993ecb79c13e81b68da43a596499f) file? I now added and extended that information in the user guide.

Currently the API and Contributing information are one file in the user guide. In the long term, this should probably be split. Maybe it does not hurt to do that right away.

Thinking about it, I would propose to create a separate `CONTRIBUTING.rst` file that resides in the main folder of the project and replaces the README file. This file could be integrated in the User Guide, like for [LumiSpy](https://lumispy.readthedocs.io/en/latest/contributing.html).

**Comment by ericpre** [comment]
Yes, sounds good!

File: docs/api.rst [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+* `HyperSpy <https://hyperspy.org>`_: Multidimensional data analysis 
+
+* Any `HyperSpy extension <https://github.com/hyperspy/hyperspy-extensions-list>`_
+  that inherits the IO capabilities:
+
+      * `LumiSpy <https://lumispy.org>`_: Luminescence analysis with HyperSpy
+      * `Kikuchipy <https://kikuchipy.org>`_: Processing, simulating and analyzing
+        electron backscatter diffraction (EBSD) patterns in Python 
+      * `PyXem <https://pyxem.readthedocs.io>`_: An open-source Python library for
+        multi-dimensional diffraction microscopy.
```
Comment (ericpre) [inline docs/api.rst:66]:
```suggestion
  * `LumiSpy <https://lumispy.org>`_: Luminescence analysis with HyperSpy
  * `Kikuchipy <https://kikuchipy.org>`_: Processing, simulating and analyzing
    electron backscatter diffraction (EBSD) patterns in Python 
  * `PyXem <https://pyxem.readthedocs.io>`_: An open-source Python library for
    multi-dimensional diffraction microscopy. 
```

**Review by ericpre — APPROVED** [review]

**Pull request merged**